File: app/src/main/java/org/connectbot/util/PreferenceConstants.java
/*
 * ConnectBot: simple, powerful, open-source SSH client for Android
 * Copyright 2007 Kenny Root, Jeffrey Sharkey
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.connectbot.util;

import android.os.Build;

/**
 * @author Kenny Root
 *
 */
public final class PreferenceConstants {
	private PreferenceConstants() {
	}

	public static final boolean PRE_ECLAIR = Build.VERSION.SDK_INT < 5;
	public static final boolean PRE_FROYO = Build.VERSION.SDK_INT < 8;

	public static final String MEMKEYS = "memkeys";

	public static final String SCROLLBACK = "scrollback";

	public static final String EMULATION = "emulation";

	public static final String ROTATION = "rotation";

	public static final String BACKUP_KEYS = "backupkeys";
	public static final boolean BACKUP_KEYS_DEFAULT = false;

	public static final String ROTATION_DEFAULT = "Default";
	public static final String ROTATION_LANDSCAPE = "Force landscape";
	public static final String ROTATION_PORTRAIT = "Force portrait";

	public static final String FULLSCREEN = "fullscreen";
	public static final String TITLEBARHIDE = "titlebarhide";
	public static final String PG_UPDN_GESTURE = "pgupdngesture";

	public static final String KEYMODE = "keymode";
	public static final String KEY_ALWAYS_VISIBLE = "alwaysvisible";
	public static final String KEY_MAP = "keymap";

	public static final String KEYMODE_RIGHT = "Use right-side keys";
	public static final String KEYMODE_LEFT = "Use left-side keys";
	public static final String KEYMODE_NONE = "none";

	public static final String CAMERA = "camera";

	public static final String CAMERA_CTRLA_SPACE = "Ctrl+A then Space";
	public static final String CAMERA_CTRLA = "Ctrl+A";
	public static final String CAMERA_ESC = "Esc";
	public static final String CAMERA_ESC_A = "Esc+A";

	public static final String KEEP_ALIVE = "keepalive";

	public static final String WIFI_LOCK = "wifilock";

	public static final String BUMPY_ARROWS = "bumpyarrows";

	public static final String SORT_BY_COLOR = "sortByColor";

	public static final String BELL = "bell";
	public static final String BELL_VOLUME = "bellVolume";
	public static final String BELL_VIBRATE = "bellVibrate";
	public static final String BELL_NOTIFICATION = "bellNotification";
	public static final float DEFAULT_BELL_VOLUME = 0.25f;

	public static final String CONNECTION_PERSIST = "connPersist";

	public static final String SHIFT_FKEYS = "shiftfkeys";
	public static final String CTRL_FKEYS = "ctrlfkeys";
	public static final String VOLUME_FONT = "volumefont";
	public static final String STICKY_MODIFIERS = "stickymodifiers";
	public static final String YES = "yes";
	public static final String NO = "no";
	public static final String ALT = "alt";

	/* Backup identifiers */
	public static final String BACKUP_PREF_KEY = "prefs";
}

File: app/src/main/java/org/connectbot/util/PreferenceConstants.kt
/*
 * ConnectBot: simple, powerful, open-source SSH client for Android
 * Copyright 2007 Kenny Root
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.connectbot.util

/**
 * @author Kenny Root
 */
object PreferenceConstants {
    const val MEMKEYS: String = "memkeys"

    const val SCROLLBACK: String = "scrollback"

    const val EMULATION: String = "emulation"

    const val ROTATION: String = "rotation"

    const val BACKUP_KEYS: String = "backupkeys"
    const val BACKUP_KEYS_DEFAULT: Boolean = false

    const val ROTATION_DEFAULT: String = "Default"
    const val ROTATION_LANDSCAPE: String = "Force landscape"
    const val ROTATION_PORTRAIT: String = "Force portrait"

    const val FULLSCREEN: String = "fullscreen"
    const val TITLEBARHIDE: String = "titlebarhide"
    const val PG_UPDN_GESTURE: String = "pgupdngesture"

    const val KEYMODE: String = "keymode"
    const val KEY_ALWAYS_VISIBLE: String = "alwaysvisible"

    const val KEYMODE_RIGHT: String = "Use right-side keys"
    const val KEYMODE_LEFT: String = "Use left-side keys"
    const val KEYMODE_NONE: String = "none"

    const val CAMERA: String = "camera"

    const val CAMERA_CTRLA_SPACE: String = "Ctrl+A then Space"
    const val CAMERA_CTRLA: String = "Ctrl+A"
    const val CAMERA_ESC: String = "Esc"
    const val CAMERA_ESC_A: String = "Esc+A"

    const val KEEP_ALIVE: String = "keepalive"

    const val WIFI_LOCK: String = "wifilock"

    const val BUMPY_ARROWS: String = "bumpyarrows"

    const val SORT_BY_COLOR: String = "sortByColor"

    const val BELL: String = "bell"
    const val BELL_VOLUME: String = "bellVolume"
    const val BELL_VIBRATE: String = "bellVibrate"
    const val BELL_NOTIFICATION: String = "bellNotification"
    const val DEFAULT_BELL_VOLUME: Float = 0.25f

    const val CONNECTION_PERSIST: String = "connPersist"

    const val SHIFT_FKEYS: String = "shiftfkeys"
    const val CTRL_FKEYS: String = "ctrlfkeys"
    const val VOLUME_FONT: String = "volumefont"
    const val STICKY_MODIFIERS: String = "stickymodifiers"
    const val YES: String = "yes"
    const val NO: String = "no"
    const val ALT: String = "alt"

    /* Backup identifiers */
    const val BACKUP_PREF_KEY: String = "prefs"
}
